packages/flutter/lib/src/material/material_state.dart

```diff
@@ -391,6 +393,203 @@ abstract class MaterialStateOutlinedBorder extends OutlinedBorder implements Mat
   OutlinedBorder? resolve(Set<MaterialState> states);
 }
 
+
+/// Defines a [TextStyle] that is also a [MaterialStateProperty].
+///
+/// This class exists to enable widgets with [TextStyle] valued properties
+/// to also accept [MaterialStateProperty<TextStyle>] values. A material
+/// state text style property represents a text style which depends on
+/// a widget's "interactive state". This state is represented as a
+/// [Set] of [MaterialState]s, like [MaterialState.pressed],
+/// [MaterialState.focused] and [MaterialState.hovered].
+///
+/// To use a [MaterialStateTextStyle], you can either:
+///   1. Create a subclass of [MaterialStateTextStyle] and implement the abstract `resolve` method.
+///   2. Use [MaterialStateTextStyle.resolveWith] and pass in a callback that
+///      will be used to resolve the color in the given states.
+///
+/// If a [MaterialStateTextStyle] is used for a property or a parameter that doesn't
+/// support resolving [MaterialStateProperty<TextStyle>]s, then its default color
+/// value will be used for all states.
+///
+/// To define a `const` [MaterialStateTextStyle], you'll need to extend
+/// [MaterialStateTextStyle] and override its [resolve] method. You'll also need
+/// to provide a `defaultValue` to the super constructor, so that we can know
+/// at compile-time what its default color is.
+///
+/// This class enables existing widget implementations with [TextStyle]
```

Comment:
Could this paragraph be merged with the first one?  Or maybe just leave out the 1st sentence here.

Same comment for the other new MaterialStateFoo classes.

Reply:
This text is a copy paste text from the MaterialStateColor class with minor modifications, I could. change it, but then it probably should be changed there as well?

Reply:
Yes and thank you.

Reply:
I think the first sentence ```This class enables existing widget implementations with [Color] properties to be extended to also effectively support `MaterialStateProperty<Color>` property values```is redundant as it is already described in the first paragraph, the second line ```[MaterialStateColor] should only be used with widgets that document their support, like [TimePickerThemeData.dayPeriodColor].```I wonder if is important enough to stand out as its's own paragraph.